jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py

```diff
@@ -180,6 +188,148 @@ def get_template_output_name_mapping(config: JobbergateConfig, job_name: str) ->
+    """
+
+    Abort.require_condition(template_path.is_file(), f"Template file {template_path} does not exist or is not a file")
+
+    with open(template_path, "r") as f:
+        file_content = f.read()
+
+    with Abort.handle_errors(
+        f"Unable to process jinja template filename={template_path}",
+        raise_kwargs=dict(
+            subject="Unable to process jinja template",
+            log_message=f"Unable to process jinja template filename={template_path}",
+        ),
+    ):
+        template = Template(file_content)
+
+    render_contexts = [parameters, {"data": parameters}]
+
+    for context in render_contexts:
+        try:
+            return template.render(**context)
+        except UndefinedError as e:
+            logger.debug(
+                "Unable to render filename={} with context={} -- Error: {}",
+                template_path,
+                context,
+                str(e),
+            )
+
+    raise Abort(
+        f"Unable to render filename={template} with context={context}",
+        subject="Unable to render jinja template",
+        log_message=f"Unable to render filename={template} with context={context}",
+    )
+
+
+def render_job_script_locally(
+    jg_ctx: JobbergateContext,
+    job_script_name: str,
+    application_path: pathlib.Path,
+    output_path: pathlib.Path,
+    sbatch_params: Optional[List[str]] = None,
+    param_file: Optional[pathlib.Path] = None,
+    fast: bool = False,
+):
+    """
+    Render a new job script from an application in a local directory.
+
+    :param str job_script_name: Name of the new job script.
+    :param pathlib.Path application_path: Path to the base application.
+    :param pathlib.Path output_path: Path to the output the rendered job script.
+    :param Optional[List[str]] sbatch_params: List of sbatch parameters.
+    :param Optional[pathlib.Path] param_file: Path to a parameters file.
+    :param bool fast: Whether to use default answers (when available) instead of asking the user.
+    :param JobbergateContext jg_ctx: The Jobbergate context.
+    :return JobScriptResponse: The new job script.
+    """
+    # Make static type checkers happy
+    assert jg_ctx.client is not None
+
+    app_data = fetch_application_data_locally(application_path)
+
+    if not app_data.workflow_files:
+        raise Abort(
+            "Application does not have a workflow file",
+            subject="Workflow file not found",
+            log_message="Application does not have a workflow file",
+        )
+
+    application_source_code = app_data.workflow_files[0].path.read_text()
+
+    (app_config, app_module) = load_application_data(app_data, application_source_code)
+
+    supplied_params = validate_parameter_file(param_file) if param_file else dict()
+    execute_application(app_module, app_config, supplied_params, fast_mode=fast)
+    update_template_files_information(app_data, app_config)
+
+    param_dict_flat = flatten_param_dict(app_config.dict())
```

Comment:
The dynamic name is missing here, in case the application code sets it at runtime.

```suggestion
    param_dict_flat = flatten_param_dict(app_config.dict())

    if param_dict_flat.get("job_script_name"):
        # Possibly overwrite script name if set at runtime by the application
        job_script_name = param_dict_flat["job_script_name"]
        logger.debug("Job script name was set by the application at runtime: {}", job_script_name)
```

Reply:
Good catch, I added it.